vars/mavenExecute.groovy

```diff
@@ -117,8 +123,17 @@ void call(Map parameters = [:]) {
         if (defines?.trim()){
             command += " ${defines}"
         }
-        dockerExecute(script: script, dockerImage: configuration.dockerImage, dockerOptions: configuration.dockerOptions) {
-            sh command
+        if (configuration.returnStdout) {
+            dockerExecute(script: script, dockerImage: configuration.dockerImage, dockerOptions: configuration.dockerOptions) {
+                commandOutput = sh(returnStdout: true, script: command)
+            }
+        } else {
+            dockerExecute(script: script, dockerImage: configuration.dockerImage, dockerOptions: configuration.dockerOptions) {
+                sh command
+            }
         }
     }
+    if (configuration.returnStdout) {
+        return commandOutput
```

Comment:
1.) This will not work when we use the step inside a declarative pipelines. It will work when using the step from other steps. In case of a declarative pipeline we cannot assign the return value to a variable.

In the light of declarative pipelines we should be careful with exceptions here. In case we go for an exception here (... allowing a return value here) we should clearly document under what circumstances the step does not work in declarative pipelines.

Normally we handle this via writing the `commonPipelineEnvironment` instances associated with the pipeline script.

2.) In case of `configuration.returnStdout` we return the `commandOutput`. In the other case there is no defined return value. There is not no return value in this case. The result of the latest executed statement will be returned - whatever this is. Maybe it would be better to always return something what we explicitly define, maybe `null`.

Reply:
Is it like we cannot use the step at all in declarative pipelines if we return a value or just that we cannot assign it?

Reply:
Like already written above:

> In case of a declarative pipeline we cannot assign the return value to a variable.

We can use the step in declarative pipelines, but we cannot assign the return value to any variable.